apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/SamhsaMatcherR4FromClaimTransformerV2Test.java

```diff
@@ -236,26 +237,34 @@ public SamhsaMatcherR4FromClaimTransformerV2Test(
 
   /**
    * Verifies that when an item has multiple unknown codings and none are HCPCS, the matcher will
-   * return true since the code system of one of the entries was unknown. This is a safety feature
-   * to ensure we filter if we don't know the system, just in case.
+   * return an exception since at least one code needs to be HCPCS. This generally indicates a
+   * coding error as there should always be a HCPCS code.
    */
   @Test
   public void
-      testR4SamhsaMatcherWhenTransformedClaimHasItemWithMultipleCodesAndAllUnknownCodesExpectMatch() {
+      testR4SamhsaMatcherWhenTransformedClaimHasItemWithMultipleCodesAndAllUnknownCodesExpectException() {
     boolean expectMatch = true;
 
     // PDE has no SAMHSA, so expect no match on SAMSHA filter
     if (claim instanceof PartDEvent) {
       expectMatch = false;
     }
 
-    verifySamhsaMatcherForItemWithMultiCoding(
-        "unknown/system/code",
-        NON_SAMHSA_HCPCS_CODE,
-        "unknown/system/code",
-        NON_SAMHSA_HCPCS_CODE,
-        expectMatch,
-        loadedExplanationOfBenefit);
+    try {
+      verifySamhsaMatcherForItemWithMultiCoding(
+          "unknown/system/code",
+          NON_SAMHSA_HCPCS_CODE,
+          "unknown/system/code",
+          NON_SAMHSA_HCPCS_CODE,
+          expectMatch,
+          loadedExplanationOfBenefit);
+    } catch (IllegalArgumentException ex) {
+      // Expect an exception; dont use junit built in because some paths dont hit an exception
+      return;
+    }
```

Comment:
Please see my clarification on this. I'm guessing, though, that we _are_ expecting exceptions to occur now? If so, we should likely revert that change.

Reply:
Adjusted back to false, apologies for the misunderstanding!